pjmedia/src/pjmedia/sdp.c

```diff
@@ -1433,6 +1431,7 @@ PJ_DEF(pj_status_t) pjmedia_sdp_parse( pj_pool_t *pool,
     }
     PJ_CATCH_ANY {
 	
+	ctx.last_error = PJ_GET_EXCEPTION();
```

Comment:
Some functions throw `SYNTAX_ERROR`, so `ctx.last_error` may be assigned  to `SYNTAX_ERROR` and returned by `pjmedia_sdp_parse()`, while app will expect the function to return a PJMEDIA error code.

I guess the alternatives are:
- change all throws to throw proper PJMEDIA error code, or
- using volatile approach (please test first), `parse_context` var in the function must be volatile, also all functions accepting and modifying `parse_context` param, must declare it as volatile too, e.g:
   ```
   static void parse_media(pj_scanner *scanner, pjmedia_sdp_media *med,
			volatile parse_context *ctx)
   ```

Reply:
The issue is reproducible by specifying `-O2` on the `CFLAGS` and assertion raised when there's a SDP parse error (exception raised). Modification using volatile approach eliminate the assertion and make sure the error code was set correctly.
- Tested using clang (macOS) with and without -O2 CFLAGS
- Tested using Visual Studio